src/models/leaderboard.js

```diff
@@ -13,29 +13,26 @@ const leaderboardSchema = new mongoose.Schema({
 })
 
 leaderboardSchema.statics.update = async () => {
-    User.find({}).populate({
+    const users = await User.find({}).populate({
         path: "stocksOwned.stock",
         model: "Stock",
         select: "pricePerUnit"
-    }).then(async (users) => {
+    })
 
-        users.forEach(user => {
-            let stockPrice = 0
-            user.stocksOwned.forEach(stock => {
```

Comment:
Change the variable name from stock to stockOwned here. Come on man, follow the convention.